### PR TITLE
Keyboard state module

### PR DESCRIPTION
**Pull request by GrantMoyer**
Adds a new module which shows the state of num lock, caps lock, and scroll lock.

screencap

### CODE REVIEW

**Comment by Alexays** [comment]
Hi, Can you add this module into man?

**Comment by GrantMoyer** [comment]
It looks like I need to fix the BSD build.

**Comment by GrantMoyer** [comment]
The build should pass on all test systems now (it passes in my [staging PR][1]). There are a few outdated github actions workflows queued up for this PR, and I'm not sure GitHub will cancel them automatically, so you may want to cancel those manually.

[1]: https://github.com/GrantMoyer/Waybar/pull/1

File: src/modules/keyboard_state.cpp [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+#include <spdlog/spdlog.h>
+
+extern "C" {
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+}
+
+waybar::modules::KeyboardState::KeyboardState(const std::string& id, const Bar& bar, const Json::Value& config)
+    : AModule(config, "keyboard_state", id, false, !config["disable-scroll"].asBool()),
```
Comment (Alexays) [inline src/modules/keyboard_state.cpp:12]: Prefer keyboard-state over keyboard_state

File: src/factory.cpp [hunk 2]
```diff
@@ -70,6 +70,11 @@ waybar::AModule* waybar::Factory::makeModule(const std::string& name) const {
       return new waybar::modules::Backlight(id, config_[name]);
     }
 #endif
+#ifdef HAVE_LIBEVDEV
+    if (ref == "keyboard_state") {
```
Comment (Alexays) [inline src/factory.cpp:74]: Prefer keyboard-state over keyboard_state

File: resources/style.css [hunk 3]
```diff
@@ -228,3 +228,19 @@ label:focus {
     margin: 0 5px;
     min-width: 16px;
 }
+
+#keyboard_state {
```
Comment (Alexays) [inline resources/style.css:232]: Prefer keyboard-state over keyboard_state

File: resources/config [hunk 4]
```diff
@@ -23,6 +23,15 @@
     //         "default": ""
     //     }
     // },
+    "keyboard_state": {
```
Comment (Alexays) [inline resources/config:26]: Prefer keyboard-state over keyboard_state

File: man/waybar-keyboard-state.5.scd [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+The following *format-icons* can be set.
+
+- *locked*: Will be shown when the keyboard state is locked. Default "locked".
+- *unlocked*: Will be shown when the keyboard state is not locked. Default "unlocked"
+
+# EXAMPLE:
+
+```
+"keyboard_state": {
```
Comment (Alexays) [inline man/waybar-keyboard-state.5.scd:64]: Prefer keyboard-state over keyboard_state

**Review by Alexays — CHANGES_REQUESTED** [review]

**Comment by Alexays** [comment]
Thanks!

**Pull request merged**